### DISCUSSION TITLE
Cannot bind argument to parameter 'ResourceKey' because it is an empty string

### DISCUSSION THREAD

**Discussion by jlasdev** [Esri/arcgis-powershell-dsc#498] [Q&A]
Hi,

I came across an issue during the configuration of the Data Store and I do not understand why, the error messages are not helping me to get this solved. Am I forgetting a specific parameter? I have already checked and re-checked but I could not find any misalignment with the templates. Could you please help me?

The template file + error messages are attached. 

It is important to mention that I can configure the DataStore manually.

`Dot Sourcing the Configuration:- ArcGISDataStore
PSDesiredStateConfiguration\Node : Cannot bind argument to parameter 'ResourceKey' because it is an empty string.
At C:\Program Files\WindowsPowerShell\Modules\ArcGIS\Configurations-OnPrem\ArcGISDataStore.ps1:86 char:5`

`PSDesiredStateConfiguration\Node : An exception was raised while processing Node 'WEB01': Index operation failed; the array index evaluated to null.
At C:\Program Files\WindowsPowerShell\Modules\ArcGIS\Configurations-OnPrem\ArcGISDataStore.ps1:86 char:5`

`Starting DSC Job:- ArcGISDataStore
Start-DscConfiguration : The computer name was not specified and the configuration directory does not have any configuration files.
At C:\Program Files\WindowsPowerShell\Modules\ArcGIS\ArcGIS.psm1:256 char:16`

[DataStoreErrorMessages.txt](https://github.com/Esri/arcgis-powershell-dsc/files/12917342/DataStoreErrorMessages.txt)

[Template.json](https://github.com/Esri/arcgis-powershell-dsc/files/12917380/Template.json)

**Comment by cameronkroeker** [comment]
Hi @jlasdev,

Which version of ArcGIS Module is being used?

Thanks,
Cameron K.

**Comment by vishalgupta-microsoft** [comment]
Hi @jlasdev  did you get a resolution for this? I'm facing same issue in my configuration block which is different from the ArcGISDataStore, so seems like a generic issue.

**Reply by jlasdev** [reply]
I can't exactly remember what the problem was, but it was definitely something I was doing wrong. 
I remember I was still learning how to use the DSC Module and I was facing some communication issues between machines, so by solving that I was able to execute the script from an "orchestrator" machine and all the other ones started receiving commands. Another mistake I can recall is that: lets say, for machine ABC I wanted to install Portal and for machine DEF I wanted to install the server, I remember only leaving the block I intended to install for a given machine. So for ABC I only had Portal and for DEF I only had the server, which was **wrong**, because the same file must be available across all the machines, at least, based on my experience.

If you need some extra info, just lemme know
